docs: expand on allocation GC details (#26792)
hashicorp/nomad: website/content/docs/configuration/client.mdx

- `gc_interval` `(string: "1m")` - Specifies the interval at which Nomad
  attempts to garbage collect terminal allocation directories.

- `gc_disk_usage_threshold` `(float: 80)` - Specifies the disk usage percent
  which Nomad tries to maintain by garbage collecting terminal allocations. Note
  that Nomad immediately garbage collects terminal allocations if garbage
  collected on the server.

- `gc_inode_usage_threshold` `(float: 70)` - Specifies the inode usage percent
  which Nomad tries to maintain by garbage collecting terminal allocations. Note
  that Nomad immediately garbage collects terminal allocations if garbage
  collected on the server.

- `gc_max_allocs` `(int: 50)` - Specifies the maximum number of allocations
  which a client will track before triggering a garbage collection of terminal
  allocations. This will _not_ limit the number of allocations a node can run at
  a time, however after `gc_max_allocs` every new allocation will cause terminal

>   allocations to be GC'd. Note that Nomad immediately garbage collects terminal allocations if garbage collected on the server.

Reviewer: more line wrap

```suggestion
  allocations to be GC'd. Note that Nomad immediately garbage collects terminal
  allocations if garbage collected on the server.
```

- `gc_interval` `(string: "1m")` - Specifies the interval at which Nomad
  attempts to garbage collect terminal allocation directories.

- `gc_disk_usage_threshold` `(float: 80)` - Specifies the disk usage percent
  which Nomad tries to maintain by garbage collecting terminal allocations. Note

>   that terminal allocations will be garbage collected immediately if garbage
>   collected on the server.

Reviewer 2: ```suggestion
  that Nomad immediately garbage collects terminal allocations if garbage
  collected on the server.
```
style nit update (active voice, present tense)

- `gc_inode_usage_threshold` `(float: 70)` - Specifies the inode usage percent
  which Nomad tries to maintain by garbage collecting terminal allocations. Note

>   that terminal allocations will be garbage collected immediately if garbage
>   collected on the server.

Reviewer 2: ```suggestion
  that Nomad immediately garbage collects terminal allocations if garbage
  collected on the server.
```

- `gc_max_allocs` `(int: 50)` - Specifies the maximum number of allocations
  which a client will track before triggering a garbage collection of terminal
  allocations. This will _not_ limit the number of allocations a node can run at
  a time, however after `gc_max_allocs` every new allocation will cause terminal

>   allocations to be GC'd. Note that terminal allocations will be garbage
>   collected immediately if garbage collected on the server.

Reviewer 2: ```suggestion
  allocations to be GC'd. Note that Nomad immediately garbage collects terminal allocations if garbage collected on the server.
```
